vllm/v1/worker/gpu_model_runner.py

```diff
@@ -944,6 +945,44 @@ def execute_model(
         sample_hidden_states = hidden_states[logits_indices]
         logits = self.model.compute_logits(sample_hidden_states, None)
 
+        # Apply guided decoding bitmasks if present
+        grammar_bitmask = scheduler_output.grammar_bitmask
+        if grammar_bitmask is not None:
+            grammar_bitmask = torch.from_numpy(grammar_bitmask)
```

Comment:
Generally, Numpy is faster than PyTorch CPU tensors. Please use Numpy, and only convert it to torch Tensor when necessary.

Reply:
This is done. I delayed converting back to a tensor until absolutely necessary